subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -209,46 +219,83 @@ public void execute(String originRepo) {
+                task.setPullRequestDescription(getPullRequestDescription(upgradeDependencyExtension));
+                task.setUpstreamRepositoryName(conf.getGitHub().getRepository());
 
                 gitOriginPlugin.provideOriginRepo(task, new Action<String>() {
                     @Override
                     public void execute(String originRepoName) {
                         task.setForkRepositoryName(originRepoName);
-                        task.setUpstreamRepositoryName(conf.getGitHub().getRepository());
                     }
                 });
 
-                findOpenPullRequestTask.provideBranchTo(task, new Action<String>() {
+                findOpenPullRequestTask.provideOpenPullRequest(task, new Action<OpenPullRequest>() {
                     @Override
-                    public void execute(String openPullRequestBranch) {
+                    public void execute(OpenPullRequest openPullRequestBranch) {
                         task.setVersionBranch(getCurrentVersionBranchName(upgradeDependencyExtension.getDependencyName(),
-                            upgradeDependencyExtension.getNewVersion(), openPullRequestBranch));
+                            upgradeDependencyExtension.getNewVersion(), openPullRequestBranch.getRef()));
+                    }
+                });
+
+                task.onlyIf(wasOpenPullRequestNotFound(findOpenPullRequestTask));
+                task.onlyIf(wasBuildFileUpdatedSpec(replaceVersionTask));
+            }
+        });
+
+        TaskMaker.task(project, MERGE_PULL_REQUEST, MergePullRequestTask.class, new Action<MergePullRequestTask>() {
+            @Override
+            public void execute(final MergePullRequestTask task) {
+                task.setDescription("Merge pull request when all checks will be passed");
+                task.mustRunAfter(CREATE_PULL_REQUEST);
+                task.setGitHubApiUrl(conf.getGitHub().getApiUrl());
+                task.setDryRun(conf.isDryRun());
+                task.setAuthToken(conf.getLenient().getGitHub().getWriteAuthToken());
+                task.setBaseBranch(upgradeDependencyExtension.getBaseBranch());
+                task.setUpstreamRepositoryName(conf.getGitHub().getRepository());
+
+                gitOriginPlugin.provideOriginRepo(task, new Action<String>() {
+                    @Override
+                    public void execute(String originRepoName) {
+                        task.setForkRepositoryName(originRepoName);
                     }
                 });
 
+                createPullRequestTask.provideCreatedPullRequest(task, new Action<OpenPullRequest>() {
+                    @Override
+                    public void execute(OpenPullRequest openPullRequest) {
+                        if (openPullRequest != null) {
+                            setPullRequestDataToTask(openPullRequest, task);
+                        }
+                    }
+                });
+
+                task.onlyIf(isPullRequestOpen(task));
```

Comment:
I suggest that we simplify this complexity. We can drop all 3 'onlyIf' clauses from here. Instead we add and 'else' section to line 294. Am I correct? The less complexity the better!

Reply:
@mockitoguy In the previous version, there was the usage of  PullRequest in two cases: when PR was freshly created and if it was already opened in past but the task was rerun. After discussion in some previous comments with @wwilk we decided, for now, to just try merging directly after creating a pull request. In the majority of cases a failure of the checks will force manual intervention of committers and as this is the last step in the process (at least for now), they can do merging by themselves. 

That's why this three onlyIfs were added (but, in truth, we can rid off ```task.onlyIf(isPullRequestOpen(task))``` one, as this situation should never happen in a standard case).